util/slice/slice.go

```diff
@@ -0,0 +1,28 @@
+package slice
+
+func RemoveString(slice []string, element string) []string {
+	n := len(slice)
+	if n == 1 {
+		return []string{}
+	}
+	for i, v := range slice {
+		if element == v {
+			if n-2 > i {
+				slice = append(slice[:i], slice[i+1:]...)
+			} else {
+				slice = slice[:i]
+			}
```

Comment:
This is unnecessary. You can actually index `slice[i:]` even if `i >= len(slice)`. Take a look at https://play.golang.org/p/30-L5CiKhLe